File: 0768-max-chunks-to-make-sorted-ii/0768-max-chunks-to-make-sorted-ii.java
class Solution {
    public int maxChunksToSorted(int[] arr) {
        int[] sortedArr = arr.clone();
        Arrays.sort(sortedArr);
        int ans = 0, sortedSum = 0, currSum = 0;
        for (int i = 0; i < arr.length; i++) {
            sortedSum += sortedArr[i];
            currSum += arr[i];
            if (sortedSum == currSum) {
                ans += 1;
            }
        }
        return ans;
    }
}

File: 0768-max-chunks-to-make-sorted-ii/0768-max-chunks-to-make-sorted-ii.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        minFromEnd, chunks = arr[:], 1
        for i in range(len(arr) - 2, -1, -1):
            if minFromEnd[i + 1] < arr[i]:
                minFromEnd[i] = minFromEnd[i + 1]
        currMax = arr[0]
        for i in range(len(arr) - 1):
            currMax = max(arr[i], currMax)
            if currMax <= minFromEnd[i + 1]:
                chunks += 1
        return chunks